File: src/main/resources/output/IFC4X3_python/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcGroup/IfcSystem/IfcBuiltSystem/IfcBuiltSystem.py
class IfcBuiltSystem(IfcSystem):
    def __init__(self):
        self.PredefinedType: IfcBuiltSystemTypeEnum = None
        self.LongName: IfcLabel = None

    # === WHERE CLAUSES ===
    # CorrectPredefinedType : NOT(EXISTS(PredefinedType)) OR (PredefinedType <> IfcBuiltSystemTypeEnum.USERDEFINED) OR ((PredefinedType = IfcBuiltSystemTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcObject.ObjectType))

File: src/main/resources/output/IFC4X3_csharp/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcGroup/IfcSystem/IfcBuiltSystem/IfcBuiltSystem.cs
public class IfcBuiltSystem : IfcSystem
{
    public IfcBuiltSystemTypeEnum PredefinedType { get; set; }
    public IfcLabel LongName { get; set; }

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : NOT(EXISTS(PredefinedType)) OR (PredefinedType <> IfcBuiltSystemTypeEnum.USERDEFINED) OR ((PredefinedType = IfcBuiltSystemTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcObject.ObjectType))
}
